WIP: update_service: Add docs for the OpenShift Update Service (#26219)
openshift/openshift-docs: modules/update-service-install-cli.adoc

[id="update-service-install-cli_{context}"]

> = Installing the {product-title} Update Service by using the CLI

Reviewer: The session is introducing the way to install the update service operator but the title is saying installing the Update Service. I would suggest to change it to Installing the {product-title} Update Service operator by using the CLI

Reviewer 2: Agree. I've introduced the sections headers with either `Operator` or `application` to distinguish between operator and operand. This PR is obsolete and will be closed. Please see https://github.com/openshift/openshift-docs/pull/29630 for latest.

You can use the {product-title} CLI to install the {product-title} Update Service Operator.

.Procedure

To install the {product-title} Update Service Operator by using the {product-title} CLI:

. Create a Namespace for the {product-title} Update Service Operator:

.. Create a Namespace object YAML file, for example, `update-service-namespace.yaml`, for the {product-title} Update Service Operator:
+
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:

>   name: FIXME-default-namespace

Author: This should probably be [`openshift-updateservice`][1].

[1]: https://github.com/openshift/cincinnati-operator/blob/4dd95332da8a79f8516255ebb54af0c84baffe72/config/olm-catalog/updateservice-operator/1.0.0/updateservice-operator.v1.0.0.clusterserviceversion.yaml#L23

  annotations:
    openshift.io/node-selector: ""
  labels:
    openshift.io/cluster-monitoring: "true" <1>
----
<1> Set the `openshift.io/cluster-monitoring` label to enable operator recommended cluster monitoring on this namespace.

.. Create the Namespace:
+
[source,terminal]
----
$ oc create -f <file-name>.yaml
----
+
For example:
+
[source,terminal]
----
$ oc create -f update-service-namespace.yaml
----

> . Install the {product-title} Update Service Operator by creating the following objects:

Reviewer 2: From my testing, if these are created separately you must create the OperatorGroup before the Subscription or the operator does not deploy.

.. Create an Operator Group object YAML file, for example, `update-service-operator-group.yaml`:
+
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: update-service-operator-group
spec:
  targetNamespaces:
  - FIXME-default-operand-namespace
----

.. Create an Operator Group object:
+
[source,terminal]
----
$ oc -n FIXME-default-namespace create -f <file-name>.yaml
----
+
For example:
+
[source,terminal]
----
$ oc -n FIXME-default-namespace create -f update-service-operator-group.yaml
----

.. Create a Subscription object YAML file, for example, `update-service-subscription.yaml`:
+
.Example Subscription
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: update-service-subscription
spec:
  channel: "{product-version}" <1>
  installPlanApproval: "Automatic"
  source: "redhat-operators" <2>
  sourceNamespace: "openshift-marketplace"
  name: "FIXME-update-service-operator"
----
<1> Specify `{product-version}` as the channel.

> <2> Specify the name of the Operator channel. For clusters that do not use a custom Operator Lifecycle Manager (OLM), specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the CatalogSource object created when you configured the Operator Lifecycle Manager (OLM).

Reviewer 2: Shouldn't this
> Specify the name of the Operator channel.

be

`Specify the name of the CatalogSource that provides the Operator.`

.. Create the Subscription object:
+
[source,terminal]
----
$ oc create -f <file-name>.yaml
----
+
For example:
+
[source,terminal]
----
$ oc -n FIXME-default-namespace create -f update-service-subscription.yaml
----
+
The Update Service Operator is installed to the `FIXME-default-namespace` Namespace and targets the `FIXME-default-operand-namespace` namespace.

. Verify the Operator installation:
+
[source,terminal]
----
$ oc -n FIXME-default-operand-namespace get clusterserviceversions
----
+
.Example output
[source,terminal]
----
NAMESPACE       NAME                                            DISPLAY                  VERSION                 REPLACES   PHASE
default         elasticsearch-operator.4.6.0-202007012112.p0    Elasticsearch Operator   4.6.0-202007012112.p0              Succeeded

> FIXME

Reviewer 3: I couldn't install the Operator in this way, so I can't update this output.

> [id="update-service-install-cli"]

Author: Done with adf5ca0732 -> 2c77ff9ce2

= Installing the {product-title} Update Service using the CLI

> include::modules/common-attributes.adoc[]

Author: Done with adf5ca0732 -> 2c77ff9ce2

> You can use the {product-title} CLI to install the {product-title} Update Service operator.

Author: Done with adf5ca0732 -> 2c77ff9ce2

> == Procedure

Author: Done with adf5ca0732 -> 2c77ff9ce2

> To install the {product-title} Update Service operator using the {product-title} CLI:

Author: Done with adf5ca0732 -> 2c77ff9ce2

. Create a Namespace for the {product-title} Update Service operator:

> .. Create a Namespace object YAML file (for example, `update-service-namespace.yaml`) for the {product-title} Update Service operator:

Author: Done withadf5ca0732 -> 2c77ff9ce2

+
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: FIXME-default-namespace
  annotations:
    openshift.io/node-selector: ""
  labels:
    openshift.io/cluster-monitoring: "true" <1>
----
<1> Set the `openshift.io/cluster-monitoring` label to enable operator recommended cluster monitoring on this namespace.

.. Create the Namespace:
+
[source,terminal]
----
$ oc create -f <file-name>.yaml
----
+
For example:
+
[source,terminal]
----
$ oc create -f update-service-namespace.yaml
----

. Install the {product-title} Update Service operator by creating the following objects:

> .. Create an Operator Group object YAML file (for example, `update-service-operator-group.yaml`):

Author: Done with adf5ca0732 -> 2c77ff9ce2

+
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: update-service-operator-group
spec:
  targetNamespaces:
  - FIXME-default-operand-namespace
----

.. Create an Operator Group object:
+
[source,terminal]
----
$ oc -n FIXME-default-namespace create -f <file-name>.yaml
----
+
For example:
+
[source,terminal]
----
$ oc -n FIXME-default-namespace create -f update-service-operator-group.yaml
----

> .. Create a Subscription object YAML file (for example, `update-service-subscription.yaml`):

Author: Done with adf5ca0732 -> 2c77ff9ce2

+
.Example Subscription
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: update-service-subscription
spec:
  channel: "{product-version}" <1>
  installPlanApproval: "Automatic"
  source: "redhat-operators" <2>
  sourceNamespace: "openshift-marketplace"
  name: "FIXME-update-service-operator"
----

> <1> Specify `{product-version}` as the channel.

Author: I'm not sure, but this is cribbed from:

```console
$ git --no-pager grep 'Specify.*as the channel' origin/master
origin/master:modules/cluster-logging-deploy-cli.adoc:<2> Specify `4.6` as the channel.
origin/master:modules/cluster-logging-deploy-cli.adoc:<2> Specify `4.6` as the channel.
origin/master:modules/metering-install-operator.adoc:<3> Specify {product-version} as the channel.
```

> <1> Specify `{product-version}` as the channel.

Reviewer 2: Since we're not tied to the OCP version do we want to instead make this OSUS Operator version, i.e. v1?

> <1> Specify `{product-version}` as the channel.

Reviewer 3: @wking, did you decide on this?

> <1> Specify `{product-version}` as the channel.

Reviewer 2: `v1` is the only`channel` we have now.

> <2> Specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the CatalogSource object created when you configured the Operator Lifecycle Manager (OLM).

Reviewer 3: I think you need to explain what this is and the two options, something like:

```suggestion
<2> Specify the name of the Operator channel. For clusters that do not use a custom Operator Lifecycle Manager (OLM), specify `redhat-operators`. If your {product-title} cluster is installed on a restricted network, also known as a disconnected cluster, specify the name of the CatalogSource object created when you configured the OLM.
```

If it's not really simple to find this CatalogSource object name, I strongly suggest adding a note to the OLM configuration steps that says that you're going to need that name when you upgrade.

Author: Landed your hunk with adf5ca0732 -> 2c77ff9ce2.  I'm not sure about the "If it's not really simple..." bit.

.. Create the Subscription object:
+
[source,terminal]
----
$ oc create -f <file-name>.yaml
----
+
For example:
+
[source,terminal]
----
$ oc -n FIXME-default-namespace create -f update-service-subscription.yaml
----
+

> The Elasticsearch Operator is installed to the `FIXME-default-namespace` Namespace and targets the `FIXME-default-operand-namespace` namespace.

Reviewer 3: What?

Author: Copy/pasta.  Fixed `s/Elasticsearch/Update Service/` with adf5ca0732 -> 2c77ff9ce2

.. Verify the Operator installation:
+
[source,terminal]
----
$ oc -n FIXME-default-operand-namespace get clusterserviceversions
----
+
.Example output
[source,terminal]
----
NAMESPACE                                               NAME                                            DISPLAY                  VERSION               REPLACES   PHASE
default                                                 elasticsearch-operator.4.6.0-202007012112.p0    Elasticsearch Operator   4.6.0-202007012112.p0               Succeeded
FIXME
...
----
+

> There should be an {product-title} Update Service operator. The version number might be different than shown.

Reviewer 3: ```suggestion
If the {product-title} Update Service Operator is listed, installation succeeded. The version number might be different than shown.
```

Author: Done with adf5ca0732 -> 2c77ff9ce2
